toolchain/sem_ir/inst_kind.def

```diff
@@ -11,14 +11,31 @@
 // - CARBON_SEM_IR_INST_KIND(Name)
 //   Invoked for each kind of semantic instruction.
 //
-// The invocation of the above macro will be wrapped in one of these, which by
-// default expand to their argument:
+// The invocation of the above macro will be wrapped in one macro from each of
+// the following sets, which by default expand to their argument:
+//
+// Whether the instruction can define a type:
 // - CARBON_SEM_IR_INST_KIND_TYPE(...)
-//   Invoked for each instruction that might define a type constant.
+//   Invoked for each instruction that is always of type `type`, and might
+//   define a type constant.
+// - CARBON_SEM_IR_INST_KIND_MAYBE_TYPE(...)
+//   Invoked for each instruction that might define a type constant if it's of
+//   type `type`, but isn't necessarily of type `type`.
```

Comment:
The current "isn't necessarily of type `type`" phrasing feels slightly odd to me. e.g., if the type constant is of type `type`, wouldn't the type be `type`? Is it needed, or can we say similar without it?

```suggestion
//   Invoked for each instruction that might define a type constant if it's of
//   type `type`.
```

or more closely echoing above phrasing:

```suggestion
//   Invoked for each instruction that is *sometimes* of type `type`, and might define a a type constant.
```

Reply:
Sure:
```suggestion
//   Invoked for each instruction that is sometimes of type `type`, and might define a type constant.
```